dio/lib/src/options.dart

```diff
@@ -141,9 +149,7 @@ class BaseOptions extends _RequestConfig with OptionsMixin {
     RequestEncoder? requestEncoder,
     ResponseDecoder? responseDecoder,
     ListFormat? listFormat,
-  })  : assert(connectTimeout == null || !connectTimeout.isNegative),
```

Comment:
Did you want to get remove the connect timeout assert without replacement?

Reply:
It's already covered by the setter: https://github.com/cfug/dio/blob/37ab8abd50e89180f7e2e9a9cf1c416cf854b924/dio/lib/src/options.dart#L114-L119